src/api/File.js

```diff
@@ -202,6 +204,85 @@ class File extends Item {
         }
     }
 
+    /**
+     * Get the thumbnail url for a given BoxItem.  The function will attempt
+     * to fetch a jpg thumbnail of the given dimensions.  If this fails, the function
+     * will attempt to fectch a 1024x1024 png of the first page of the file as a fallback.
+     * (Certain file types do not have 'thumbnails' but do have 'pages', so the png fallback
+     * covers these types.)
+     *
+     * @param {BoxItem} item - item whose thumbnail should be fetched
+     * @param {string} dimensions - desired dimensions of jpg thumbnail. Acceptable dimensions
+     * for the jpg are: "32x32", "94x94", "160x160", "320x320", "1024x1024", "2048x2048".
+     * @param {Function} successCallback - function to call with the thumbnail url. The function
+     *  will recieve null as its argument if a thumbnail could not be fetched.
+     * @param {Function} errorCallback - Function to call with errors
+     * @return {void}
+     */
+    getFileThumbnail(
+        item: BoxItem,
+        dimensions: string,
```

Comment:
Do we plan to request different dimensions when the grid slider is implemented? If so, it seems like it'd be more efficient to re-use the 1024x1024 images and scale them up/down. In that case, the additional complexity here can be avoided.